modules/katamari/src/main/ts/ephox/katamari/api/Singleton.ts

```diff
@@ -50,6 +50,8 @@ export const destroyable = <T extends { destroy: () => void }> (): Revocable<T>
 
 export const unbindable = <T extends { unbind: () => void }> (): Revocable<T> => singleton<T>((s) => s.unbind());
 
+export const repeatable = (): Revocable<number> => singleton((intervalId) => clearInterval(intervalId));
```

Comment:
I wonder if the fact that it's an interval can be moved inside the `repeatable`.

I'm thinking create it as `Singleton.repeatable(ScrollIntervalValue)` and then `interval.set(functionToRepeat)`. That would be a pretty cool data type, and avoid all the manual interval management.